boot-qemu.sh

```diff
@@ -223,8 +220,16 @@ function setup_qemu_args() {
     esac
     checkbin "${QEMU[*]}"
 
-    [[ ${KIMAGE:=zImage} == "vmlinux" ]] || BOOT_DIR=arch/${ARCH}/boot/
-    KERNEL=${KBUILD_DIR}/${BOOT_DIR}${KIMAGE}
+    # If '-k' is an path that ends in the kernel image, we can just use it directly
+    if [[ ${KERNEL_LOCATION##*/} = "${KIMAGE:=zImage}" ]]; then
```

Comment:
what is `##*/`?
---
Do none of the existing arch's specify a `KIMAGE` of `zImage`? (I assume there's probably a list of possible image names, since the compression scheme is encoded in the name).

Reply:
`##*/` is bash parameter substitution for `basename` essentially (remove the last `/` and everything before it): https://wiki.bash-hackers.org/syntax/pe#from_the_beginning

No, because `zImage` used to be the most common image name. All of `ARCH=arm` uses it and 32-bit PowerPC used it prior to 2a68826b5af13c326577f45c156ff399fc202a21). I should probably just add `KIMAGE` to all of the arm32 switches to avoid that cryptic assignment but I do not feel strongly about it.